apiserver/src/main/resources/migration/procedures/procedure_migrate_metrics_to_partitions.sql

```diff
@@ -0,0 +1,53 @@
+CREATE OR REPLACE PROCEDURE "MIGRATE_METRICS_TO_PARTITIONS"(
+    target_table TEXT,
+    source_table TEXT
+)
+LANGUAGE "plpgsql"
+AS
+$$
+DECLARE
+    retention_days INTEGER := 90;
+    start_date DATE;
+    end_date DATE := current_date;
+    partition_date DATE;
+    partition_name TEXT;
+    next_day DATE;
+BEGIN
+    -- Fetch retention value from config
+    SELECT COALESCE(cp."PROPERTYVALUE"::INTEGER, 90)
+    INTO retention_days
+    FROM "CONFIGPROPERTY" cp
+    WHERE cp."GROUPNAME" = 'maintenance'
+      AND cp."PROPERTYNAME" = 'metrics.retention.days';
+
+    start_date := current_date - retention_days;
```

Comment:
Nit: Let's make this more explicit. I had to check first if subtracting an integer from a date even has the intended effect 😅 

```suggestion
    start_date := current_date - (retention_days * '1 day'::interval);
```